src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -1,117 +1,294 @@
+    //   the names listed in `commandNames`, and `keys` is an array of
+    //   keypresses. Currently, the array should contain a single string, which
+    //   can be deserialized into a KeyChord.
+    // Arguments:
+    // - json: and arrayof JsonObject's to deserialize into our _keyShortcuts mapping.
+    // Return Value:
+    // - the newly constructed AppKeyBindings object.
+    TerminalApp::AppKeyBindings AppKeyBindings::FromJson(Windows::Data::Json::JsonArray const& json)
+    {
+        TerminalApp::AppKeyBindings newBindings{};
+
+        for (const auto& value : json)
+        {
+            if (value.ValueType() == JsonValueType::Object)
+            {
+                JsonObject obj = value.GetObjectW();
+                if (obj.HasKey(COMMAND_KEY) && obj.HasKey(KEYS_KEY))
+                {
+                    const auto commandString = obj.GetNamedString(COMMAND_KEY);
+                    const auto keys = obj.GetNamedArray(KEYS_KEY);
+                    if (keys.Size() != 1)
+                    {
+                        continue;
+                    }
+                    const auto keyChordString = keys.GetAt(0).GetString();
+                    ShortcutAction action;
+                    bool parsedSuccessfully = false;
+                    // Try matching the command to one we have
+                    for (const auto& pair : commandNames)
+                    {
+                        if (pair.second == commandString)
+                        {
+                            action = pair.first;
+                            parsedSuccessfully = true;
+                            break;
+                        }
+                    }
+                    if (!parsedSuccessfully)
+                    {
+                        continue;
+                    }
+                    // Try parsing the chord
+                    Settings::KeyChord chord{ false, false, false, 0 };
+                    parsedSuccessfully = false;
+                    try
+                    {
+                        chord = Settings::KeyChord::FromString(keyChordString);
+                        parsedSuccessfully = true;
+                    }
+                    catch (...)
+                    {
+                        continue;
+                    }
+                    if (parsedSuccessfully)
+                    {
+                        newBindings.SetKeyBinding(action, chord);
+                    }
+                }
+            }
+        }
+        return newBindings;
+    }
+
+    // Function Description:
+    // - Small helper to insert a given KeyChord, ShortcutAction pair into the
+    //   given json array
+    // Arguments:
+    // - bindingsArray: The JsonArray to insert the object into.
+    // - chord: The KeyChord to serailize and place in the json array
+    // - actionName: the name of the ShortcutAction to use with this KeyChord
+    void _AddShortcutToJsonArray(JsonArray bindingsArray,

```

Comment:
should `bindingsArray` be a const reF? it _does not need to_ take ownership, i think

Reply:
Why can it be _const_? I'm pretty clearly appending to it, but cppwinrt clearly doesn't care